src/test/java/com/metamx/common/guava/ConcatSequenceTest.java

```diff
@@ -187,6 +190,85 @@ public void cleanup(Iterator<Sequence<Integer>> iterFromMake)
     SequenceTestHelper.testClosed(closedCount, seq);
   }
 
+  @Test
+  public void testEnsureNextSequenceIsCalledLazilyInToYielder() throws Exception
+  {
+    final AtomicBoolean lastSeqFullyRead = new AtomicBoolean(true);
+
+    Sequence<Integer> seq = Sequences.concat(
+        Sequences.map(
+            Sequences.simple(
+                ImmutableList.of(
+                    ImmutableList.of(1, 2, 3),
+                    ImmutableList.of(4, 5, 6)
+                )
+            ),
+            new Function<ImmutableList<Integer>, Sequence<Integer>>()
+            {
+              @Override
+              public Sequence<Integer> apply(final ImmutableList<Integer> input)
+              {
+                if (lastSeqFullyRead.getAndSet(false)) {
+                  return Sequences.simple(
+                      new Iterable<Integer>()
+                      {
+                        private Iterator<Integer> baseIter = input.iterator();
+
+                        @Override
+                        public Iterator<Integer> iterator()
```

Comment:
You're coding in java8 aren't you :) this is missing some stuff that's causing the unit tests to fail in java7